src/main/java/com/vamae/authorization/service/UserService.java

```diff
@@ -15,7 +15,14 @@ public class UserService {
 
     public int changeBalance(String username, int offset) {
         User user = findUserByUsername(username);
-        user.setBalance(user.getBalance() + offset);
+        int userBalance = user.getBalance();
+        if((offset < 0 && userBalance >= -offset) || offset > 0) {
```

Comment:
You can left only "if(offset < 0 && userBalance >= -offset)" but change "userBalance >=" to "userBalance <="
And just throw exception in its block